trieste/acquisition/function/active_learning.py

```diff
@@ -0,0 +1,245 @@
+# Copyright 2021 The Trieste Contributors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+This module contains acquisition function builders and acquisition functions for Bayesian active
```

Comment:
Maybe we should discuss if active learning is the right term. We should use active learning as an umbrella term, which contains BO. But I'm not sure what to call this sub-class then...

Reply:
the community somehow divides these problems into BO and BAL, but I agree BAL is probably more general and BO a special case

for now I would probably stick to active_learning and mull it over until we see how to go about it, as it seems like using BAL as an umbrella would entail a broader refactoring and change of language etc